cmd/mixpanel_to_redshift.go

```diff
@@ -51,14 +57,30 @@ func main() {
 	}
 
 	if *copyToRedshift {
-		r, err := redshift.NewRedshift()
+		s3Info := redshift.S3Info{
+			Region:    env.MustGet("AWS_REGION"),
+			AccessID:  env.MustGet("AWS_ACCESS_KEY_ID"),
+			SecretKey: env.MustGet("AWS_SECRET_ACCESS_KEY"),
+		}
+		r, err := redshift.NewRedshift(*host, *port, *db, *user, *password, *redshiftTimeout, s3Info)
 		defer r.Close()
 		if err != nil {
 			log.Fatal(err)
 		}
-		if err := r.CopyJSONDataFromS3(*redshiftSchema, *redshiftTable, exportFile, *jsonpathsFile, awsRegion); err != nil {
+		// do use s3 creds, don't use GZIP
+		// I am not confident this currently works - you may have to set timeformat to 'epochsecs'
+		tx, err := r.Begin()
+		if err != nil {
+			log.Fatal(err)
+		}
+		if err := r.RunJSONCopy(tx, *schema, *table, exportFile, *jsonpathsFile, true, false); err != nil {
+			tx.Rollback()
 			log.Fatal(err)
 		}
+		if err := tx.Commit(); err != nil {
+			log.Fatal(err)
+		}
+
 		if err := r.VacuumAnalyze(); err != nil {
```

Comment:
Shouldn't need this since we are now using `STATUPDATE ON` and `VACUUM`s are very disruptive.